glib-macros/tests/test.rs

```diff
@@ -279,6 +279,59 @@ fn subclassable() {
     }
 }
 
+#[test]
+fn subclass_derives_name() {
+    mod foo {
+        use glib::subclass::prelude::*;
+
+        mod imp {
+            use super::*;
+
+            #[derive(Default)]
+            pub struct Foo {}
+
+            #[glib::object_subclass]
+            impl ObjectSubclass for Foo {
+                type Type = super::Foo;
+            }
+
+            impl ObjectImpl for Foo {}
+        }
+
+        glib::wrapper! {
+            pub struct Foo(ObjectSubclass<imp::Foo>);
+        }
+    }
+
+    mod bar {
+        use glib::subclass::prelude::*;
+
+        mod imp {
+            use super::*;
+
+            #[derive(Default)]
+            pub struct Bar {}
+
+            #[glib::object_subclass]
+            impl ObjectSubclass for super::imp::Bar {
```

Comment:
`super::imp::Bar` is redundant but shows how name is derived. It is `SuperImpBar` and not just a `Bar`.

A reason to reject this PR?

Reply:
No opinion here but that name would be displayed for cases like GtkWidgets in the inspector. Having all the names prefixed with SuperImp or Imp would be confusing. I guess we could strip those two prefixes?